packages/next/src/server/lib/incremental-cache/index.ts

```diff
@@ -556,7 +559,11 @@ export class IncrementalCache implements IncrementalCacheType {
 
     if (this.dev && !ctx.fetchCache) return
     // fetchCache has upper limit of 2MB per-entry currently
-    if (ctx.fetchCache && JSON.stringify(data).length > 2 * 1024 * 1024) {
+    if (
+      ctx.fetchCache &&
+      this.isFetchCacherPossible &&
```

Comment:
```suggestion
      // we don't show this error/warning when a custom cache handler is being used
      // as it might not have this limit
      !this.hasCustomCacheHandler &&
```
Can we tweak this to a boolean that signals if the user provided a custom cache handler instead?

Reply:
“I have created a flag that determines whether the use of `FetchCache` is possible at least in theory - if no custom provider is passed, and additionally configured it so that it is not an implementation of `FetchCache` as a protection against special individuals (*like me :)*).”

`this.isFetchCacherPossible = !CurCacheHandler || CurCacheHandler instanceof FetchCache`

The name `hasCustomCacheHandler` only signifies the first part (`!CurCacheHandler`), but is not suitable for inheritance (`CurCacheHandler instanceof FetchCache`). Maybe, the second part is overthinking and can be ignored. If that's case, I can rename it.

Reply:
Yeah that second check isn't needed as `CurCacheHandler` is only provided when we are using a custom handler other than our built-in ones.

Reply:
Okay, thank you! Fixed.

Also made a slight correction to the comment above (*`fetchCache has upper limit of 2MB per-entry currently` -> `FetchCache has upper limit of 2MB per-entry currently`*), because in this context, fetchCache means that fetch is currently cached, but the limit is provided by the same named provider (*maybe it wasn't the best naming choice*).